### PR TITLE
fix(createURL): correctly remove page in state

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

createURL didn't reset the page because it uses SearchParameter methods vs. helper methods which get the page reset automatically.

**Result**

fixes https://github.com/algolia/vue-instantsearch/issues/932

all createURLs now reset the page as expected

### CODE REVIEW

**Comment by Haroenv** [comment]
In the future we should rely on getWidgetSearchParameters IMO, although at the moment that doesn't reset the page either (otherwise it doesn't work when pagination or infinite hits isn't the last widget). Maybe `isPageReset` could be useful somehow?

**Comment by jonathan-bird** [comment]
@Haroenv That was pretty much my approach too when I created my hacky workaround but thinking about it more, you want pretty much every facet in the URL except pagination so I think a "reset pagination" type method is the best way to handle it.

**Comment by eunjae-lee** [comment]
> In the future we should rely on getWidgetSearchParameters IMO, although at the moment that doesn't reset the page either (otherwise it doesn't work when pagination or infinite hits isn't the last widget). Maybe `isPageReset` could be useful somehow? Or call `this.getWidgetSearchParameters().resetPage()` in refine and createURL, which could make sense too?

While this PR looks good to me, could you rephrase what you said?

**Comment by Haroenv** [comment]
If you check the implementations of getWidgetSearchParameters, you'll see that none of them reset the page. That's because the actual page reset logic is done in the index widget's change listener on the helper.

That means that even if we use getWidgetSearchParameters for createURL and refine, we still would have to reset the page after retrieving the state.

It could be quite interesting longer-term to use a createURL based on UIState alone though, as well as relying on uiState alone for refine. I think it should be possible in theory, but will have breaking changes for non-builtin widgets relying only on the helper state. Maybe in a major version?

**Review by eunjae-lee — APPROVED** [review]
Thanks for the explanation. This PR looks good to me.
